client/src/pages/course/[taskDetails].tsx

```diff
@@ -0,0 +1,294 @@
+import React, { useMemo, useState } from 'react';
+
+import { Header } from 'components';
+import { useRouter } from 'next/router'
+import { useAsync } from 'react-use';
+import {
+  Row,
+  Col,
+  Typography,
+  Tooltip,
+} from 'antd';
+
+import { EventTypeColor, EventTypeToName } from 'components/Schedule/model';
+
+import {
+  renderTag,
+} from 'components/Table';
+
+import withCourseData from 'components/withCourseData';
+import { withSession } from 'components';
+import { useLoading } from 'components/useLoading';
+
+import { CourseEvent, CourseService, CourseTaskDetails } from '../../services/course';
+import { CoursePageProps } from 'services/models';
+
+enum OccasionTypes {
+  UNKNOWN = 'UNKNOWN',
+  EVENT = 'EVENT',
+  TASK = 'TASK',
+}
+
+export function TaskDetailsPage(props: CoursePageProps) {
+  const router = useRouter();
+  const { taskDetails } = router.query;
+
+  const [occasion, setOccasion] = useState<CourseTaskDetails | CourseEvent>();
+  const [occasionType, setOccasionType] = useState(OccasionTypes.UNKNOWN);
+  const [, withLoading] = useLoading(false);
+
+  const courseService = useMemo(() => new CourseService(props.course.id), [props.course.id]);
+
+  useAsync(
+    withLoading(async () => {
+      const [events, tasks] = await Promise.all([
+        courseService.getCourseEvents(),
+        courseService.getCourseTasksDetails(),
+      ]);
+
+      const occasion1 = events.filter(({ id }) => id === Number(taskDetails))[0];
+      if (occasion1) {
+        setOccasion(occasion1);
+        setOccasionType(OccasionTypes.EVENT);
+        return;
+      }
+
+      const occasion2 = tasks.filter(({ id }) => id === Number(taskDetails))[0];
+      if (occasion2) {
+        setOccasion(occasion2)
+        setOccasionType(OccasionTypes.TASK);
+      }
+    }),
+    [courseService],
+  );
+  const { Title, Text, /* Link, Paragraph */ } = Typography;
+
+  if (occasionType === OccasionTypes.TASK) {
```

Comment:
So, since the logic is pretty similar, we can use a query param to determine the type of the event, and based on this info decide what to render. I suggest to create three components: 
1. Dumb component that renders event details
2. Dumb component that renders task details
3. Smart component that handles all logic and renders a right view based on query param

Reply:
ok, i created two dumb pages with different routes (instead dumb components), but instead of smart component i handle logic on the router level in tableView component. It takes few lines and i think that we don't required smart component in this case.